terraform/modules/application/events.tf

```diff
@@ -0,0 +1,43 @@
+resource "aws_cloudwatch_event_rule" "app_task_exited" {
+  name = "${var.prefix}-app-task-exited"
+
+  event_pattern = jsonencode(
+    {
+      "source" : [
+        "aws.ecs"
+      ],
+      "detail-type" : [
+        "ECS Task State Change"
+      ],
+      "detail" : {
+        "group" : [
+          "service:${aws_ecs_service.app.name}"
+        ],
+        "stoppedReason" : [
+          "Essential container in task exited"
+        ]
+      }
+    }
+  )
+}
+
+resource "aws_cloudwatch_event_rule" "app_service_action_problem" {
+  name = "${var.prefix}-app-service-action-problem"
+
+  event_pattern = jsonencode(
+    {
+      "source" : [
+        "aws.ecs"
+      ],
+      "detail-type" : [
+        "ECS Service Action"
+      ],
+      "resources" : [
+        aws_ecs_service.app.name
```

Comment:
Looks like this may need the `arn` instead (see [here](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events-structure.html)), and may be why we haven't seen it triggered it when the ECS was failing to launch tasks y'day during my experimenting

I only got to this link by looking at `event_pattern` terraform docs [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_event_rule#event_pattern), which sends you to [here](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-events.html), which then has a link to the above (Event structure reference) 😅